app/models/UserAnswers.scala

```diff
@@ -65,7 +67,28 @@ case class UserAnswers(
       None
     else 
       Some(set(questionPage, newValue).get)
-  
+
+  /** If user's answer has changed, passes updated user-answers object to given save function  
+    * @param questionPage - the user-answer we might be changing
+    * @param newValue - the user's answer
+    * @param saveUserAnswerFunc - function to call if answer has changed
+    * @param format - formatter for user's answer object type
+    * @tparam A - type of user's answer
+    * @return
+    *  - Future of false if user's answer is the same as the current value
+    *  - Future of true if user's answer has changed
+    */
+  def change_v3[A] (questionPage: QuestionPage[A], newValue: A, saveUserAnswerFunc: SaveUserAnswerFunc) 
```

Comment:
Can we give this a more descriptive name please?

Reply:
Sure - it will replace change()